src/DotNetWorker.Core/Context/DefaultFunctionContext.cs

```diff
@@ -12,13 +13,15 @@ internal sealed class DefaultFunctionContext : FunctionContext, IAsyncDisposable
     {
         private readonly IServiceScopeFactory _serviceScopeFactory;
         private readonly FunctionInvocation _invocation;
+        private readonly CancellationToken _cancellationToken;
 
         private IServiceScope? _instanceServicesScope;
         private IServiceProvider? _instanceServices;
         private BindingContext? _bindingContext;
 
-        public DefaultFunctionContext(IServiceScopeFactory serviceScopeFactory, IInvocationFeatures features)
+        public DefaultFunctionContext(IServiceScopeFactory serviceScopeFactory, IInvocationFeatures features, CancellationToken cancellationToken)
         {
+            _cancellationToken = cancellationToken;
             _serviceScopeFactory = serviceScopeFactory ?? throw new ArgumentNullException(nameof(serviceScopeFactory));
             Features = features ?? throw new ArgumentNullException(nameof(features));
 
```

Comment:
We've established a pattern of pulling properties directly from `IInvocationFeatures`... should se pull `CancellationToken` from there a well and not have it passed in directly?

Reply:
I'm not super familiar with `IInvocationFeatures`, but I do remember @fabiocav mentioning that we shouldn't use features for this - is this the same thing? If the correct pattern is for us to use a feature for this then happy to do that

Reply:
Had a sync with Fabio and we think we won't get much benefit introducing the cancellation token